keel-core/src/main/kotlin/com/netflix/spinnaker/keel/core/api/EnvironmentArtifactPin.kt

```diff
@@ -9,7 +9,15 @@ data class EnvironmentArtifactPin(
   val version: String,
   val pinnedBy: String?,
   val comment: String?
-)
+) {
+  init {
+    if (comment != null) {
+      require(comment.length <= 255) {
+        "Comment length should be 255 characters max."
```

Comment:
This can be displayed in the UI, correct? In that case, I'd suggest tweaking the message a bit:
```suggestion
        "Comments should have a maximum size of 255 characters."
```